### PR TITLE
Add association type endpoint

### LINKED ISSUE #276: Simplify associations and facet counts with association_type field
The latest version of monarch contains an association_type field.  This could simplify some of the calls in biolink, especially when generating counts and for gene-gene associations.

### PR DESCRIPTION
**Pull request by kshefchek**
Adds generic association type endpoint (for monarch solr), for example /api/association/type/gene_function?fetch_objects=true&start=0&rows=25&unselect_evidence=true&subject=HGNC:3603

closes https://github.com/biolink/biolink-api/issues/276

### CODE REVIEW

File: biolink/api/link/endpoints/associations_from.py [hunk 1]
```diff
@@ -67,3 +67,23 @@ def get(self, subject, object):
         """
         args = core_parser.parse_args()
         return search_associations(subject=subject, object=object, user_agent=USER_AGENT, **args)
+
+@api.doc(params={'association_type': 'Association type, eg gene_phenotype'})
```
Comment (cmungall) [inline biolink/api/link/endpoints/associations_from.py:71]: will you also support subject.cat and object.cat (to avoid forcing string concatenation)?

**Review by cmungall — APPROVED** [review]
cool!

**Review by deepakunni3 — APPROVED** [review]

File: biolink/api/link/endpoints/associations_from.py (hunk 1 above)
Comment (kshefchek) [reply biolink/api/link/endpoints/associations_from.py:71]: I added [association_type](https://github.com/monarch-initiative/monarch-app/commit/ff749344cfe912317fab47c1c0c30653dcd61143#diff-afd371b82f73195d2b80e578810731d7) to the monarch solr 6 months ago to help with facet counting, mostly to disambiguate gene-gene associations, this endpoint queries this field directly.  There's a subject_cat, object_cat endpoint already, /association/find/{subject_category}/{object_category}

**Comment by kshefchek** [comment]
I noticed we use "category" in the serializers code and "categories" in the biomodel/core code, resulting in category being null in some cases in API responses, I've switched categories to category in core.py to fix this but curious if theres a preference for one over the other

**Comment by deepakunni3** [comment]
@kshefchek Yes, the preference is `category` instead of `categories`.
I did try to fix it earlier, but looks like I might have missed a few places in the code.

**Comment by cmungall** [comment]
Correct. http://w3id.org/biolink/vocab/category

**Pull request merged**